tests/foreman/ui/test_ansible.py

```diff
@@ -565,6 +565,54 @@ def test_positive_ansible_config_report_changes_notice_and_failed_tasks_errors(
             assert f'notice Install the {package_name} package' in notice_log['permission_denied']
             assert f'Installed: rubygem-{package_name}' in notice_log['permission_denied']
 
+    def test_positive_non_admin_hostgroup_permission(
+        self, request, function_org, function_location, target_sat
+    ):
+        """Verify Hostgroup page is accessible to non-admin user.
+
+        :id: 3b8ce895-9d4c-491d-a5eb-d292748134a5
+
+        :customerscenario: true
+
+        :BZ: 2236418
+
+        :steps:
+            1. Create a user with non-admin permissions and assign "Ansible Roles Manager" and "Ansible Tower Inventory Reader"
+            to the user.
+            2. Create a hostgroup and add in same organization and location as user.
+            3. Login to Satellite WebUI with user credentials and navigate to Configure -> Hostgroups
+
+        :expectedresults: The hostgroup created should be visible to non-admin user.
+        """
+        password = settings.server.admin_password
+
+        ansible_manager_role = (
+            target_sat.api.Role().search(query={'search': 'name="Ansible Roles Manager"'})[0].id
+        )
+        tower_inventory_manager_role = (
+            target_sat.api.Role()
+            .search(query={'search': 'name="Ansible Tower Inventory Reader"'})[0]
+            .id
+        )
+        SELECTED_ROLE = [ansible_manager_role, tower_inventory_manager_role]
+        user = target_sat.api.User(
+            role=SELECTED_ROLE,
+            admin=False,
+            login=gen_string('alphanumeric'),
+            password=password,
+            organization=[function_org],
+            location=[function_location],
+        ).create()
+        request.addfinalizer(user.delete)
+        user_cfg = user_nailgun_config(user.login, password)
+
+        hg_name = target_sat.api.HostGroup(
+            organization=[function_org],
+            location=[function_location],
+        ).create()
+        with target_sat.ui_session(user=user_cfg.auth[0], password=password) as session:
+            assert session.hostgroup.search(hg_name.name)[0]['Name'] == hg_name.name
```

Comment:
I think this test could be parametrized with test_positive_assign_and_remove_ansible_role_to_hostgroup for admin and non-admin, as it doesn't cover any ansible scenario now, wdyt?

Reply:
Well the scenario was very specific for both tests:
First of all according to the bug, we only need 2 roles to be added to the user, that means we would need almost all the test to be added in the if section of "test_positive_assign_and_remove_ansible_role_to_hostgroup". Importing roles to the user might need different permission which is not part of bug scope and can alter bug scenario. 
That is why I would like to keep it separate with existing test.

Reply:
just to be clear, we're not testing any ansible roles here, this are user roles used which allow accessing user with this role to view hostgroup

Reply:
Based on the discussion on slack, I am resolving this conversation.